src/db-api.js

```diff
@@ -68,47 +86,143 @@ const transactionsHistoryForAddresses = (db: Pool) => async (
 
 // The remaining queries should be used only for the purposes of the legacy API!
 
-/**
- * Queries DB looking for successful transactions associated with any of the given addresses.
- * @param {Db Object} db
- * @param {Array<Address>} addresses
- */
-const bulkAddressSummary = (db: Pool) => async (addresses: Array<string>): Promise<ResultSet> =>
-  db.query({
-    text: `SELECT * FROM "txs"
-      WHERE hash = ANY (SELECT tx_hash FROM "tx_addresses" WHERE address = ANY($1))
-      AND tx_state = $2
-      ORDER BY time DESC`,
-    values: [addresses, 'Successful'],
-  })
-
 /**
 * Queries TXS table looking for a successful transaction with a given hash
 * @param {Db Object} db
-* @param {*} tx
+* @param {Transaction} tx
 */
 const txSummary = (db: Pool) => async (tx: string): Promise<ResultSet> =>
   db.query({
     text: 'SELECT * FROM "txs" WHERE hash = $1 AND tx_state = $2',
     values: [tx, 'Successful'],
   })
 
+/**
+* Queries TX table looking for tx by its hash
+* @param {Db Object} db
+* @param {Transaction} tx
+*/
+const getTx = (db: Pool) => async (tx: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT id, block, hash::text FROM "tx" WHERE hash = $1',
+    values: [tx],
+  })
+
+/**
+* Queries TX_BODY table looking for tx body by its hash
+* @param {Db Object} db
+* @param {Transaction} tx
+*/
+const getRawTx = (db: Pool) => async (tx: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT body::text as tx_body FROM tx_body WHERE hash = $1',
+    values: [tx],
+  })
+
+/**
+* Queries BLOCK table looking for block with a given id
+* @param {Db Object} db
+* @param {Block} blockId
+*/
+const getBlockById = (db: Pool) => async (blockId: string): Promise<ResultSet> =>
+  db.query({
+    text: 'SELECT time, block_no, hash::text FROM block WHERE id = $1',
+    values: [blockId],
+  })
+
+/**
+* Quries TX* tables to get txInputs for a given transaction
+* @param {Db Object} db
+* @param {Transaction} tx
+*/
+const getSingleTxInputs = (db: Pool) => async (tx: string): Promise<ResultSet> =>
+  db.query({
+    text: `SELECT
+      tx_out.address, tx_out.value
+      FROM tx_out
+      INNER JOIN tx ON tx.id = tx_out.tx_id
+      INNER JOIN tx_in ON tx_in.tx_out_id = tx_out.tx_id AND tx_in.tx_out_index = tx_out.index
+      WHERE tx_in.tx_in_id = $1`,
+    values: [tx],
+  })
+
+/**
+* Queries TX, BLOCK, TX_OUT tables to acquire inward and outward transactions for given addresses
+* @param {Db Object} db
+* @param {Array<Address>} addresses
+*/
+const getTransactions = (db: Pool) => async (addresses: Array<string>): Promise<ResultSet> =>
+  db.query({
+    text: `SELECT
```

Comment:
shouldn't this be `SELECT DISTINCT` as well?

Reply:
It was originally 1 to 1 mapping from cardano-explorer-api haskell code, but it makes sense to me to make it `DISTINCT`.